src/web/components/SlotBodyEnd.tsx

```diff
@@ -108,9 +119,21 @@ const MemoisedInner = (props: Props) => {
             );
     }, []); // only ever call once (we'd rather fail then call the API multiple times)
 
+    useEffect(() => {
```

Comment:
Is it worth lifting this into the useHasBeenSeen function as well? (or anyhow out of this component itself? It doesn't matter too much, I just wonder if it harms readability of the overall logic, which is important because of the whole memoisation/special nature it has.

Reply:
Good question. I think I'd vote to keep it in here because it's tied into the render lifecycle of this component (and it's possible in other places that we might not need to wrap logic around becoming visible in a `useEffect`). What do you think?